tiledb/sm/rest/rest_capabilities.h

```diff
@@ -0,0 +1,106 @@
+/**
+ * @file   rest_capabilities.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2025 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * Helper class to encapsulate REST supported versions and capabilities.
+ */
+
+#include "tiledb/sm/rest/rest_client.h"
+
+namespace tiledb::sm {
+
+class RestClientRemote;
+
+class RestCapabilities {
+  friend RestClientRemote;
+  using TileDBVersion = RestClient::TileDBVersion;
+
+ public:
+  /**
+   * Default constructor allows the class to be constructed without submitting
+   * a REST request to initialize member variables.
+   */
+  RestCapabilities() = default;
+
+  ~RestCapabilities() = default;
+
+  /**
+   * Fully initialized constructor contains all REST version and capabilities
+   * information required for handling edge cases between client & server
+   * releases.
+   */
+  RestCapabilities(
+      TileDBVersion rest_version, TileDBVersion rest_minimum_version);
```

Comment:
Since we now have a dedicated file for capabilities I'd also move `TileDBVersion` struct definition here so that we don't have to jump between files.

Reply:
Done